src/averageAges.js

```diff
@@ -37,7 +45,27 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWithChildren = [];
+
+  if (withChildren) {
+    people.forEach(item => {
+      if (people.filter(x => x.mother).map(x => x.mother).includes(item.name)) {
+        arrayWithChildren.push(item);
+      }
+    });
+
+    return arrayWithChildren
+      .map(x => x.died - x.born)
+      .reduce((x, y) => (x + y)) / arrayWithChildren.length;
+  }
+
+  return (people
+    .filter((x) => x.sex === 'f')
+    .map((x) => (x.died - x.born))
+    .reduce((x, y) => (x + y)))
+      / (people
+        .filter((x) => x.sex === 'f')
+        .map((x) => (x.died - x.born)).length);
```

Comment:
This logic is hard to understand. Try to find out a simpler solution. KISS - keep it simple.